packages/deploy-config-sub-generator/src/app/index.ts

```diff
@@ -0,0 +1,467 @@
+import { basename, dirname, join } from 'path';
+import dotenv from 'dotenv';
+import autocomplete from 'inquirer-autocomplete-prompt';
+import { withCondition } from '@sap-ux/inquirer-common';
+import { UI5Config } from '@sap-ux/ui5-config';
+import { FileName, getMtaPath, findCapProjectRoot } from '@sap-ux/project-access';
+import { getAbapQuestions, indexHtmlExists } from '@sap-ux/abap-deploy-config-sub-generator';
+import { ApiHubType, useAbapDirectServiceBinding } from '@sap-ux/cf-deploy-config-writer';
+import { getHostEnvironment, hostEnvironment } from '@sap-ux/fiori-generator-shared';
+import { isFullUrlDestination } from '@sap-ux/btp-utils';
+import {
+    DeploymentGenerator,
+    bail,
+    showOverwriteQuestion,
+    generateDestinationName,
+    getDestination,
+    ErrorHandler,
+    TargetName,
+    getConfirmConfigUpdatePrompt
+} from '@sap-ux/deploy-config-generator-shared';
+import { getEnvApiHubConfig, t, generatorNamespace } from '../utils';
+import {
+    getCFQuestions,
+    loadManifest,
+    API_BUSINESS_HUB_ENTERPRISE_PREFIX,
+    DESTINATION_AUTHTYPE_NOTFOUND
+} from '@sap-ux/cf-deploy-config-sub-generator';
+import { parseTarget } from './utils';
+import { getDeployTargetQuestion } from '../common/prompts';
+import type { CfDeployConfigAnswers, CfDeployConfigOptions } from '@sap-ux/cf-deploy-config-sub-generator';
+import type { CfDeployConfigQuestions } from '@sap-ux/cf-deploy-config-inquirer';
+import type { DeployConfigOptions, Target } from '../types';
+import type { AbapDeployConfigAnswersInternal, AbapDeployConfigQuestion } from '@sap-ux/abap-deploy-config-inquirer';
+import type { ApiHubConfig } from '@sap-ux/cf-deploy-config-writer';
+import type { Answers, Question } from 'inquirer';
+import type { FioriToolsProxyConfigBackend } from '@sap-ux/ui5-config';
+
+/**
+ * The main deployment configuration generator.
+ */
+export default class extends DeploymentGenerator {
+    private readonly launchDeployConfigAsSubGenerator: boolean;
+    private readonly launchStandaloneFromYui: boolean;
+    private readonly abapChoice: Target = { name: TargetName.ABAP, description: 'ABAP' };
```

Comment:
Is it correct to define inquirer related concepts in a generator constructor. Can these be moved to where they are used?

Reply:
Done